fe/fe-core/src/main/java/org/apache/doris/planner/external/ExternalFileScanNode.java

```diff
@@ -0,0 +1,432 @@
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package org.apache.doris.planner.external;
+
+import com.google.common.base.Strings;
+import org.apache.doris.analysis.Analyzer;
+import org.apache.doris.analysis.ArithmeticExpr;
+import org.apache.doris.analysis.Expr;
+import org.apache.doris.analysis.FunctionCallExpr;
+import org.apache.doris.analysis.IntLiteral;
+import org.apache.doris.analysis.NullLiteral;
+import org.apache.doris.analysis.SlotDescriptor;
+import org.apache.doris.analysis.SlotRef;
+import org.apache.doris.analysis.StringLiteral;
+import org.apache.doris.analysis.TupleDescriptor;
+import org.apache.doris.analysis.UserIdentity;
+import org.apache.doris.catalog.*;
+import org.apache.doris.catalog.Table.TableType;
+import org.apache.doris.common.AnalysisException;
+import org.apache.doris.common.Config;
+import org.apache.doris.common.DdlException;
+import org.apache.doris.common.UserException;
+import org.apache.doris.common.util.BrokerUtil;
+import org.apache.doris.mysql.privilege.UserProperty;
+
+import org.apache.doris.planner.PlanNodeId;
+import org.apache.doris.qe.ConnectContext;
+import org.apache.doris.resource.Tag;
+import org.apache.doris.system.Backend;
+import org.apache.doris.system.BeSelectionPolicy;
+import org.apache.doris.thrift.*;
+
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+import com.google.common.collect.Sets;
+import org.apache.hadoop.hive.metastore.api.FieldSchema;
+import org.apache.hadoop.hive.metastore.api.Table;
+import org.apache.hadoop.mapred.FileSplit;
+import org.apache.hadoop.mapred.InputSplit;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.mortbay.log.Log;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.*;
+
+import static org.apache.doris.planner.HiveScanNode.HIVE_DEFAULT_COLUMN_SEPARATOR;
+import static org.apache.doris.planner.HiveScanNode.HIVE_DEFAULT_LINE_DELIMITER;
+
+public class ExternalFileScanNode extends ExternalScanNode {
+    private static final Logger LOG = LogManager.getLogger(ExternalFileScanNode.class);
+
+    private static class ParamCreateContext {
+        public TBrokerScanRangeParams params;
+        public TupleDescriptor srcTupleDescriptor;
+        public Map<String, Expr> exprMap;
+        public Map<String, SlotDescriptor> slotDescByName;
+    }
+
+    private final org.apache.doris.catalog.Table catalogTable;
```

Comment:
I think it's better to use the new table interface org.apache.doris.catalog.TableIf instead. But we can change it later as well.

Reply:
Yes, now it only use it test the code of `FileScanNode`. When the `TableIf ` is ready, this pr will only support the external datasource.